R/writeNumSumReport.R

```diff
@@ -0,0 +1,67 @@
+#' Writes a rmd file for the Numeric Summary Report [WIP]
+#'
+#' @param num.sum.report [\code{NumSumReport} Object]\cr
+#'   The report Object which should be converted to an rmd file
+#' @param sub.dir [\code{character(1)}]\cr
+#'   the name of the (relative) sub-directory where the data report will be saved. Default is \code{Data_Report}
+#' @param save.mode [\code{logical(1)}]\cr
+#'   In Save mode its not possible to use an existing folder.
+#'   To ensure no data is lost, a new folder will be created (if possible).
+#'   Default is \code{TRUE}
+#' @examples
+#'  data("Boston", package = "MASS")
+#'  num.sum.task = makeNumSumTask(id = "BostonTask", data = Boston, target = "medv")
+#'  #get the numeric summary task object
+#'  num.sum = makeNumSum(num.sum.task)
+#'  #create the numeric summary report
+#'  num.sum.report = makeNumSumReport(num.sum)
+#'  #write the report
+#'  writeReport(num.sum.report)
+#' @return Invisible NULL
+#' @import checkmate
+#' @export
+writeReport.NumSumReport = function(num.sum.report, sub.dir = "Data_Report", save.mode = TRUE){
+  assertClass(num.sum.report, "NumSumReport")
+  assertCharacter(sub.dir, len = 1L, min.chars = 1L)
+  assertLogical(save.mode, len = 1L)
+  # Create sub directory, save current wd and set new wd to the new directory
+  origin.wd = createDir(sub.dir, save.mode)
+  rmd.name = rmdName("NumSumReport")
+
+  # TryCatch sets wd back and closes all open connections if an error occurs
+  tryCatch({
+    ##try part:
+    #start the report file
+    #  report.con = file(paste0("basicReport_", basic.report$report.task$dataset.name, ".rmd"), "w") #or include task.id ?
+    report.con = file(rmd.name, "w")
+
+    writeLines("## Numeric Summary Report from AEDA containing numeric summary as well as plots", con  = report.con)
+    writeLines("```{r}", con = report.con)
+    writeLines("devtools::load_all()", con = report.con)
+    writeLines("#library(AEDA)", con = report.con)
```

Comment:
In general I'm not sure where to load AEDA. I would add it to the needed packages where it really is needed the first time. Then you dont have use write lines and instead use  `rmdLibrary(needed.pkgs, report.con)` even if it is just one package. This makes the code more uniform

Reply:
will have a look later into rmdLibrary and where to use it.